content/tutorials/git_authentication/index.Rmd

```diff
@@ -0,0 +1,266 @@
+
+## TL;DR
+
+Enable two-factor authentication for your GitHub account, follow [these steps](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-totp-mobile-app).
+Another two-factor authentication app that is not mentioned on that website but easy to use is Google Authenticator.
+You probably already use a two-factor authentication app, just use that.
+
+Next, in R, install the `usethis` package (this will also install packages `gert` and `gitcreds` which we use below):
+
+```{r}
+install.packages("usethis")
+```
+
+Use the following commands to configure some global git options (if you haven't done this already):
+
+```{r eval=FALSE}
+?gert::git_config_global_set # read the help file
+gert::git_config_global_set("user.name", "Your Name") # change Your Name
+gert::git_config_global_set("user.email", "your@email.com") # change your@email.com
+gert::git_config_global_set("init.defaultBranch", "main")
+```
+
+If you plan to use functions that create something on GitHub (a repo, an issue, a pull request, a branch, ...) and for all remote operations from the command line (git pull, git push, git clone, git fetch), you will need to use the following commands to create a Personal Access Token (PAT) and add it to the Git Credential Manager: 
+
+```{r eval=FALSE}
+?usethis::git_vaccinate() # read the help file
+usethis::git_vaccinate()
+?usethis::create_github_token() # read the help file
+usethis::create_github_token() #browser opens, follow instructions
+```
+
+Add the PAT to the Git Credential Manager:
+
+```{r eval=FALSE}
+?gitcreds::gitcreds_set() # read the help file
+gitcreds::gitcreds_set() #paste PAT
+```
+
+**WARNING**:  handle your Personal Access Token (PAT) as as secret.
+Anyone who has your token, has access to your GitHub account.
+
+Check if everything is OK:
+
+```{r eval=FALSE}
+usethis::git_sitrep()
+```
+
+The default Git protocol should be 'https' (recommended on Windows).
+
+## Modes of authentication
+
+The way in which you need to authenticate depends on how you access your resources in GitHub.
+For instance, to manage your repositories online you can sign-in to GitHub using a username and password followed by two-factor authentication.
+As another example, a function like `checklist::new_branch()` will create a new branch on GitHub and locally.
+In order to create a new branch, you will need a Personal Access Token with appropriate scopes if you use the HTTPS protocol.
+A token with no assigned scopes can only access public information.
+
+We will follow the [recommendations given in the usethis package](https://usethis.r-lib.org/articles/articles/git-credentials.html) for Git authentication on Windows, which covers all ways which you will need to safely authenticate:
```

Comment:
```suggestion
We will follow the [recommendations given in the `usethis` package](https://usethis.r-lib.org/articles/articles/git-credentials.html) for safe Git authentication on Windows, which cover multiple facets:
```

Reply:
I think there are more places where a `monospaced` font is better used (function and package names in text).